packages/api/src/external/carequality/patient-demographics.ts

```diff
@@ -0,0 +1,89 @@
+import { Patient } from "@metriport/core/domain/patient";
+import { LinkDemographics } from "@metriport/core/domain/patient-demographics";
+import { PatientResource } from "@metriport/ihe-gateway-sdk";
+import {
+  scoreLink,
+  linkHasNewDemographics,
+  patientToNormalizedCoreDemographics,
+  removeInvalidArrayValues,
+  normalizeDob,
+  normalizeGender,
+  normalizeAndStringifyNames,
+  normalizeAddress,
+  stringifyAddress,
+  normalizeTelephone,
+  normalizeEmail,
+} from "../../domain/medical/patient-demographics";
+import { CQLink } from "./cq-patient-data";
+
+export function getNewDemographics(patient: Patient, links: CQLink[]): LinkDemographics[] {
+  const coreDemographics = patientToNormalizedCoreDemographics(patient);
+  const consolidatedLinkDemographics = patient.data.consolidatedLinkDemographics;
+  return getPatientResources(links)
+    .map(patientResourceToNormalizedLinkDemographics)
+    .filter(linkDemographics => scoreLink({ coreDemographics, linkDemographics })[0])
+    .filter(linkDemographics =>
+      linkHasNewDemographics({
+        coreDemographics,
+        consolidatedLinkDemographics,
+        linkDemographics,
+      })
+    );
+}
+
+export function patientResourceToNormalizedLinkDemographics(
+  patientResource: PatientResource
+): LinkDemographics {
+  const dob = normalizeDob(patientResource.birthDate);
+  const gender = normalizeGender(patientResource.gender);
+  const names = (patientResource.name ?? []).flatMap(name => {
```

Comment:
just to confirm, why are given and family names all getting collapsed into one name list? What are the ramifications of that for different people with inverted first and last names. 

James Henry vs Henry James

Reply:
re: `why are given and family names all getting collapsed into one name list` -- my impression was that we only want to compare first name / last name combinations, and not consider only last names. The Epic PDF makes no mention of last name only (although an "approximate" name might mean only a last name...).

re: `inverted first and last names` -- not sure what you mean by inverted names. We make first name, last name pairs (for CW this means unpacking both family and given arrays, for CQ family is just a string and thus means only unpacking the given arrays), and then they are normalized, which means first name and last name will be ordered in the string representation.

Reply:
I just didnt understand. Resolving